.github/workflows/compare.yaml

```diff
@@ -30,13 +30,43 @@ jobs:
     steps:
       - name: Git Checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.WORKFLOW_GIT_ACCESS_TOKEN }}
       - name: Install coursier
         uses: coursier/setup-action@v1
         with:
           apps: scala-cli
 
+      - run: echo "filename=$(./scripts/get_filename.sh)" >> "$GITHUB_ENV"
+        id: filename
```

Comment:
For the filename, I think we could use the actual version of Scala which would be unique enough, would give more context (`3.x.y-RCz-commitSHA-date-NIGHTLY`), and would allow for easy overwriting in case of repeating the build due to some internal issues. 
If we integrate with scheduled builds we can get the version similarly as here: https://github.com/VirtusLab/community-build3/blob/7d56005233c7802baa8952c14cbb0771eda6b7ef/.github/workflows/buildExecuteScheduledWeekly.yaml#LL73C11-L73C82